platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -900,3 +908,18 @@ uint16_t _odp_ipsec_sa_alloc_ipv4_id(ipsec_sa_t *ipsec_sa)
 	/* No need to convert to BE: ID just should not be duplicated */
 	return tl->next_ipv4_id++;
 }
+
+uint64_t _odp_ipsec_sa_stats_pkts(ipsec_sa_t *sa)
+{
+	uint64_t tl_pkt_quota = 0;
+	sa_thread_local_t *sa_tl;
+	int n;
+
+	/* Aggregate per thread packet counts */
+	for (n = 0; n < ODP_THREAD_COUNT_MAX; n++) {
+		sa_tl = &ipsec_sa_tbl->per_thread[n].sa[sa->ipsec_sa_idx];
+		tl_pkt_quota += sa_tl->packet_quota;
```

Comment:
While in practice this probably would not cause any problems, this is still a data race (other thread writing to the variable while we are reading it). To make things right, sa_tl->packet_quota should be an atomic variable. It can be accessed using relaxed ordering.

Reply:
Agreed. Will have this addressed in next version.